test/assocWithP.js

```diff
@@ -0,0 +1,19 @@
+const { expect } = require('chai')
+const property   = require('prop-factory')
+const always     = require('ramda/src/always')
+
+const { assocWithP } = require('..')
+
+const assigner = assocWithP('foo', always(Promise.resolve('bar')))
```

Comment:
Similarly here, don't stub a trivial function.  Create a real transform to test the behavior.